MekHQ/src/mekhq/gui/model/UnitTableModel.java

```diff
@@ -125,16 +126,80 @@ public int getAlignment(int col) {
         };
     }
 
-    public @Nullable String getTooltip(int row, int col) {
-        Unit u = getUnit(row);
-        return switch (col) {
-            case COL_STATUS -> u.isRefitting() ? u.getRefit().getDesc() : null;
-            case COL_CREW_STATE -> u.getCrewState().getToolTipText();
-            case COL_QUIRKS -> u.getQuirksList();
+    /**
+     * Returns the tooltip for the specified row and column.
+     *
+     * @param rowIndex    the index of the row
+     * @param columnIndex the index of the column
+     * @return the tooltip for the specified row and column, or {@code null} if no tooltip is available
+     */
+    public @Nullable String getTooltip(int rowIndex, int columnIndex) {
+        Unit unit = getUnit(rowIndex);
+
+        return switch (columnIndex) {
+            case COL_STATUS -> unit.isRefitting() ? unit.getRefit().getDesc() : null;
+            case COL_CREW_STATE -> unit.getCrewState().getToolTipText();
+            case COL_CREW -> getCrewTooltip(unit);
+            case COL_QUIRKS -> unit.getQuirksList();
             default -> null;
         };
     }
 
+    /**
+     * Returns the tooltip for the crew status of a given unit.
+     *
+     * @param unit the unit for which to get the crew tooltip
+     * @return the crew tooltip as an HTML string
+     */
+    static String getCrewTooltip(Unit unit) {
```

Comment:
I suppose this could be private?

Reply:
As with the other instance, I made this package-private so I can unit test it. If that assumption is wrong, please lemme know as otherwise I'm gonna be making this mistake a lot :D

Reply:
Welllll tbh I have done only little reading w.r.t. unit tests. I will read the pragmatic guide neo suggested. I really recommend J. Blochs Effective Java, it barely touches unit tests but has loads of useful insight from a guy who designed Java. Personally I follow his rule: never upgrade visibility beyond what's strictly necessary, and unit tests are not strictly necessary. Another perspective is this: Don't unit test a class' inner workings. They are likely to change often and in fine ways and then you have to change the unit test. Also, when you change the inner workings, you know you'll have to get it right within that class but you don't need to care about anything outside of it. Rather test the interface of classes, i.e. the methods that other classes can use. When you change the behavior of such a method, other classes may or may not suddenly break and you can't very well check all the code of those.

Reply:
So, based on that, a good rule of thumb would be: if the method is public or used by multiple methods disparate, then I should unit test it; if it's only used by the one method and within a single class then there is no need to Unit Test?

Reply:
I guess. On that I'm no authority. And dont forget you're doing this on your sweet spare time.

Reply:
I've opted to include a unit test for `getCrewTooltip()` due to its' complexity and the need to ensure it's reporting correctly. However, I have reworked the unit tests based on your feedback. They should be a bit more robust now and easier to maintain.